components/ILIAS/Rating/classes/class.ilRatingCategoryGUI.php

```diff
@@ -98,20 +98,30 @@ public function executeCommand(): void
         $next_class = $ilCtrl->getNextClass($this);
         $cmd = $ilCtrl->getCmd('listCategories');
 
-        if ($this->http->wrapper()->query()->has('rating_category_ordering_table_action')) {
-            $cmd = $this->http->wrapper()->query()->retrieve(
-                'rating_category_ordering_table_action',
-                $this->refinery->kindlyTo()->string()
-            );
-        }
-
         switch ($next_class) {
             default:
                 $this->$cmd();
                 break;
         }
     }
 
+    public function handleTableActions(): void
+    {
+        $cmd = $this->http->wrapper()->query()->retrieve(
+            'rating_category_ordering_table_action',
+            $this->refinery->byTrying(
+                [
+                    $this->refinery->kindlyTo()->string(),
+                    $this->refinery->always('')
+            ]
+            )
+        );
+
+        if ($cmd !== '') {
+            $this->$cmd();
```

Comment:
I suggest handling this via `switch` or `match`, otherwise every method of this class could be called.